### ISSUE TITLE
Check and add admins/moderators

### ISSUE THREAD

**Issue by xandfury** [miguelgrinberg/flasky#57]
I am trying to make a version of flasky to create my own social network. The issue I am facing is that 

``` bash
    echo  $NETWORK_ADMIN  #Equivalent to FLASK_ADMIN
```

yields the correct email address. But the account registered with that email address still does not have admin rights. Particularly  I cannot see the Edit Profile[ADMIN] {page 126 in the book} button.  

when I removed the `if current_user.is_administrator()` statement from `app/templates/user.html` I get the ``forbidden` error message on clicking the Edit Profile[ADMIN] button

How do I check which all users are the admins? I also tried querying this in the shell

```
>>> result = User.query.filter_by(email='my-email@gmail.com')
>>> result.all()
[<User u'spunkpike'>]
>>> result.role_id
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'BaseQuery' object has no attribute 'role_id'
>>> 
```

**Comment by miguelgrinberg** [comment]
To add new roles you'll have to make a few changes in the code. As it is in the book, the application supports a single admin, and everyone else get the default role. Have you added the new roles to the Role table in the database? Once you do that, you have to change the account registration route to set these new roles when a matching account is submitted for registration.

Regarding your snippet of code, if you want to check the role of a user here is the correct code:

>>> result = User.query.filter_by(email='my-email@gmail.com').first()
>>> result
<User u'spunkpike'>
>>> result.role_id

**Comment by spunkpike** [comment]
Thanks for the reply. Can you please explain this in a little more detail.
 
> Have you added the new roles to the Role table in the database? Once you do that, you have to change the account registration route to set these new roles when a matching account is submitted for registration.

In the shell : 
```shell 
>>> Role.query.all()  # yields
[<Role u'Moderator'>, <Role u'Administrator'>, <Role u'User'>]
>>> 
```

I feel stupid for the wrong code snippet. However, the corrected snippet also does not work.

```shell
>>> result = User.query.filter_by(email='my-email@gmail.com').first()
>>> result
>>> result.role_id
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'role_id'
>>> 
```

The `models.py` and `views.py` for the `auth blueprint` are the same as the 10b branch of the flasky git repo.

**Comment by miguelgrinberg** [comment]
This error:

    AttributeError: 'NoneType' object has no attribute 'role_id'

Means that the email that you used in your query was not found in the user table.

**Comment by spunkpike** [comment]
```shell
>>> result = User.query.filter_by(email='my-email@gmail.com').first()
>>> result
<User u'spunkpike'>
>>> result.role_id
>>> 
```

**Comment by miguelgrinberg** [comment]
Okay, good. Now that means that your user does not have any role set. How did you create this user?

**Comment by spunkpike** [comment]
I initially had setup the environment variable `FLASKY_ADMIN` to my email-id and then registered using the same id in registration form. This would be the form at `auth/register`

**Comment by miguelgrinberg** [comment]
So you need to add some debugging in this place: https://github.com/miguelgrinberg/flasky/blob/master/app/models.py#L123-L126

That's where the role assignment is done. Something in your environment is preventing those conditionals from selecting the right role for your user. I recommend that you delete the user and then add some logging in the area I indicated before you try the registration again.

**Comment by spunkpike** [comment]
I got it working. Like you said, virtualenv was messing with my environmental variables. I noticed this when I switched to master branch and the variables returned nothing.
 So as a temporary solution I hard-coded the variables at the end of the `venv/bin/activate` file. Everything works fine now. But the problem is that I shouldn't hard-code my `MAIL_PASSWORD` variable. 

Can you suggest and alternative solution?

**Comment by miguelgrinberg** [comment]
No, definitely you shouldn't need to hack around this.

So the problem is that the env vars were not accessible from the Python process? Did you try printing these vars from the application to see if they are coming through or not?

**Comment by spunkpike** [comment]
That seems to be the case. The application was fine so I can assume that the vars from the application were not coming through. Otherwise, there wouldn't be a problem in accessing the Edit Profile[ADMIN] button in the first place.

**Comment by keithalpichi** [comment]
I am having an issue where 

    >>> Role.query.all()  # yields an empty list
    >>> []

Why is this so? Initially my own user profile wasn't set to the admin role (I had closed Terminal and forgot to set "export FLASKY_ADMIN") so I had to set it in the shell to 1. I've even created new users but no users are given a role_id. Even doing the below outputs:

    >>> admin = User.query.filter_by(username="admin_name").first()
    >>> admin.role_id # yields
    >>> u'Administrator'
    >>> Role.query.all() # yields
    >>> []

I'm also assuming there's an issue where users aren't getting their roles set. I also thought it may be due to my database but it's all migrated and updated.
